packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/enhanced/cdi/pipeline.py

```
@@ -0,0 +1,91 @@
+    ClusteringStrategy,
+)
+
+
+@dataclass
+class CrossDocumentPipeline:
+    """Composable pipeline skeleton for CDI.
+
+    This is a non-functional scaffold to define typed extension points.
+    """
+
+    entity_extractor: Optional[EntityExtractor] = None
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)